server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java

```diff
@@ -533,15 +533,19 @@ private static Map<String, Mapper> buildMergedMappers(
             MergeReason reason,
             MapperMergeContext objectMergeContext
         ) {
-            Map<String, Mapper> mergedMappers = null;
-            for (Mapper mergeWithMapper : mergeWith) {
-                Mapper mergeIntoMapper = (mergedMappers == null ? existing.mappers : mergedMappers).get(mergeWithMapper.simpleName());
+            Iterator<Mapper> iterator = mergeWith.iterator();
+            if (iterator.hasNext() == false) {
+                return Map.copyOf(existing.mappers);
+            }
```

Comment:
This retains the behavior of the previous implementation  where the `mergedMappers` map will not by initialized if `mergeWith` is an empty iterator.

Reply:
++ this makes it explicit, it's much more readable than before.